packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -26,22 +24,33 @@ import {
   rtlTextContainer,
 } from '../../utils';
 import {
-  ShorthandRenderFunction,
-  ShorthandCollection,
   ComponentEventHandler,
   ObjectShorthandCollection,
   FluentComponentStaticProps,
+  ShorthandRenderFunction,
 } from '../../types';
-import {
-  getAllSelectableChildrenId,
-  isAllGroupChecked,
-  hasSubtree,
-  removeItemAtIndex,
-  getSiblings,
-  TreeContext,
-  TreeRenderContextValue,
-  processItemsForSelection,
-} from './utils';
+import { useSelectableTree } from './hooks/useSelectableTree';
+import { BaseFlatTreeItem } from './hooks/flattenTree';
+
+export interface TreeRenderContextValue {
+  toggleActive: (ids: string[], e: React.SyntheticEvent) => void;
+  toggleSelect: (ids: string[], e: React.SyntheticEvent) => void;
+  focusFirstChild: (itemId: string) => void;
+  focusParent: (itemId: string) => void;
+  siblingsExpand: (e: React.SyntheticEvent, itemProps: TreeItemProps) => void;
+  registerItemRef: (id: string, node: HTMLElement) => void;
+  getItemById: (id: string) => BaseFlatTreeItem & { [key: string]: any };
+}
+
+export const TreeContext = React.createContext<TreeRenderContextValue>({
+  toggleActive: _.noop,
+  toggleSelect: _.noop,
+  focusFirstChild: _.noop,
+  focusParent: _.noop,
+  siblingsExpand: _.noop,
+  registerItemRef: _.noop,
+  getItemById: id => ({ id } as any),
```

Comment:
```suggestion
  getItemById: id => ({ id } as Partial<BaseFlatTreeItem>),
```

Something like should be able to replace any...